docs/integrations/pytorch.rst

```diff
@@ -3,109 +3,370 @@
--------------------
-It is very easy to track PyTorch metrics in Neptune. The following example presents a simple integration that lets you track loss:
+If you want to try things out and focus only on the code you can either:
+
+|colab-script-neptune|
+
+.. _pytorch-quickstart:
+
+Quickstart
+----------
+This quickstart will show you how to:
+
+* Install the necessary neptune packages
+* Connect Neptune to your Optuna hyperparameter tuning code and create the first experiment
+* Log metrics, figures, and artifacts from your first Optuna sweep to Neptune, and
+* Explore them in the Neptune UI.
+
+.. _pytorch-before-you-start-basic:
+
+Before you start
+^^^^^^^^^^^^^^^^
+You have ``Python 3.x`` and following libraries installed:
+
+* ``neptune-client``. See :ref:`neptune-client installation guide <installation-neptune-client>`.
+* ``torch``. See |pytorch-install|.
+
+You also need minimal familiarity with torch. Have a look at this |pytorch-guide| to get started.
+
+.. code-block:: bash
+
+   pip install --quiet torch neptune-client
+
+Step 1: Initialize Neptune
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+Run the code below:
+
+.. code-block:: python3
+
+    import neptune
+
+    neptune.init(api_token='ANONYMOUS', project_qualified_name='shared/pytorch-integration')
+
+.. tip::
+
+    You can also use your personal API token. Read more about how to :ref:`securely set the Neptune API token <how-to-setup-api-token>`.
+
+Step 2: Create an Experiment and log parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Run the code below to create a Neptune experiment:
+
+.. code-block:: python3
+
+    neptune.create_experiment('pytorch-quickstart')
+
+This also creates a link to the experiment. Open the link in a new tab.
+The charts will currently be empty, but keep the window open. You will be able to see live metrics once logging starts.
+
+Step 3: Add logging into your training loop
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Log your loss after every batch by adding ``neptune.log_metric`` inside of the loop.
 
 .. code-block:: python3
 
     for batch_idx, (data, target) in enumerate(train_loader):
-        data, target = data.to(PARAMS['DEVICE']), target.to(PARAMS['DEVICE'])
         optimizer.zero_grad()
-        output = model(data)
-        loss = F.nll_loss(output, target)
+        outputs = model(data)
+        loss = F.nll_loss(outputs, target)
 
-        # log loss to neptune
-        neptune.log_metric('batch_loss', batch_idx, loss.data.cpu().numpy())
+        # log loss
+        neptune.log_metric('batch_loss', loss)
 
-Your loss is now logged to Neptune:
+        loss.backward()
+        optimizer.step()
+        if batch_idx == 100:
+            break
 
-.. image:: ../_static/images/integrations/ht-log-pytorch-1.png
-   :target: ../_static/images/integrations/ht-log-pytorch-1.png
-   :alt: image
+You can log epoch metric and losses by calling ``neptune.log_metric`` at the epoch level.
 
-.. note::
+Step 4: Run your training script
```

Comment:
Here I'm a bit confused.
I was not asked to create any script - I just saw snippet about where to place log_metric.

Also, in the step one you suggest to run the code. Where should I run it?